### PR TITLE
Full support of PHP 7.3 for 4.0.x

### PR DESCRIPTION
**Pull request by CameronHall**
Full support of PHP 7.3
Fixes: #13288, #12176, #12385, #12648, #12705

### CODE REVIEW

**Review by sergeyklay — APPROVED** [review]

File: tests/unit/Mvc/ViewTest.php [hunk 1]
```diff
@@ -667,7 +668,8 @@ function () {
                 // FIXME: This test need to be refactored to not use try/catch
                 try {
                     echo $a_cool_var;
-                } catch (PhpUnitException $e) {
+                    $this->fail('Variable a_cool_var is defined and is set to "' . $a_cool_var . '"');
+                } catch (\PHPUnit_Framework_Exception $e) {
```
Comment (sergeyklay) [inline tests/unit/Mvc/ViewTest.php]:
```suggestion
                } catch (\PHPUnit\Framework\Exception $e) {
```

**Review by sergeyklay — CHANGES_REQUESTED** [review]

File: tests/unit/Mvc/ViewTest.php (hunk 1 above)
Comment (sergeyklay) [reply tests/unit/Mvc/ViewTest.php]: @CameronHall It seems this is the cause of the failing test

**Comment by CameronHall** [comment]
Looks like it passed this time @sergeyklay

**Comment by sergeyklay** [comment]
Thank you

**Pull request merged**